akka-remote/src/main/scala/akka/remote/RemoteMetricsExtension.scala

```diff
@@ -54,53 +55,59 @@ private[akka] class RemoteMetricsOff extends RemoteMetrics {
   override def logPayloadBytes(msg: Any, payloadBytes: Int): Unit = ()
 }
 
+/**
+ * INTERNAL API
+ */
+private object RemoteMetricsOn {
+  val LogFrameSizeExceedingConfigKey = "akka.remote.log-frame-size-exceeding"
+}
+
 /**
  * INTERNAL API
  */
 private[akka] class RemoteMetricsOn(system: ExtendedActorSystem) extends RemoteMetrics {
 
   private val log = Logging(system, this.getClass)
 
-  private val logFrameSizeExceeding: Try[Int] = Try(
-    system.settings.config.getBytes("akka.remote.log-frame-size-exceeding").toInt
-  )
-
-  logFrameSizeExceeding.failed.foreach { e ⇒
-    val value = system.settings.config.getAnyRef("akka.remote.log-frame-size-exceeding")
-    val msg = "Caught exception of type \"" + e.getClass.getSimpleName + "\" " +
-              "while parsing \"akka.remote.log-frame-size-exceeding\". Expected a value in bytes, " +
-              "got: \"" + value + "\". For reference, check https://github.com/lightbend/config/blob/master/HOCON.md"
-    log.warning(msg)
+  private val logFrameSizeExceeding: Int = {
+    if (system.settings.config.getString(LogFrameSizeExceedingConfigKey) == "off") {
+      Int.MaxValue
+    } else {
+      try {
+        system.settings.config.getBytes(LogFrameSizeExceedingConfigKey).toInt
+      } catch {
+        case e: ConfigException ⇒
+          log.warning(e.getMessage + ". For reference, check https://github.com/lightbend/config/blob/master/HOCON.md")
+          Int.MaxValue
+      }
+    }
   }
 
   private val maxPayloadBytes: ConcurrentHashMap[Class[_], Integer] = new ConcurrentHashMap
 
   override def logPayloadBytes(msg: Any, payloadBytes: Int): Unit =
-    logFrameSizeExceeding.foreach { frameSize ⇒
-      if (payloadBytes >= frameSize) {
-        val clazz = msg match {
-          case x: ActorSelectionMessage ⇒ x.msg.getClass
-          case x: RouterEnvelope        ⇒ x.message.getClass
-          case _                        ⇒ msg.getClass
-        }
+    if (payloadBytes >= logFrameSizeExceeding) {
```

Comment:
if we put a check against Int.MaxValue first here I think the JIT can optimize away the entire block when the setting is off, since that fact will never change, instead of having to check for every element size.